resource/resource_graph.go

```diff
@@ -213,6 +213,46 @@ func (g *Graph) FindNodesByAPI(api API) []Name {
 	return ret
 }
 
+// RemoteConfigs returns the configs of all remote API nodes.
+func (g *Graph) RemoteConfigs() []Config {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	var remotes []Config
+	for k, v := range g.nodes {
+		// using client.RemoteAPI can cause an import cycle.
+		if k.API == APINamespaceRDK.WithType("remote").WithSubtype("") {
+			remotes = append(remotes, v.Config())
+		}
+	}
+	return remotes
+}
+
+// ComponentConfigs returns the configs of all component API nodes.
+func (g *Graph) ComponentConfigs() []Config {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	var components []Config
+	for k, v := range g.nodes {
+		if k.API.IsComponent() {
+			components = append(components, v.Config())
+		}
+	}
+	return components
+}
+
+// ServiceConfigs returns the configs of all service API nodes.
+func (g *Graph) ServiceConfigs() []Config {
```

Comment:
I don't think the graph should understand API types. Can we just have the robot iterate over all the resources once and do its checks?

Reply:
Good point; removed these methods.